File: Implementation/programmers/[Lv2] 이진 변환 반복하기/main.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

/*
    - Hint
    format을 대신하는 함수가 있어..? 없어... 
    그래서 숫자 -> 이진수 -> 문자열로 변환해야 할듯
    bitset<bit 상수>(숫자 변수) = 숫자 -> 이진수 
*/

vector<int> solution(string s) {
    vector<int> answer(2, 0); // 2개를 0으로 초기화 
    
    while(s != "1"){
        int count = 0;
        int n = s.size();
        for(int i=0; i<n; i++){
            if(s[i] == '1'){
                count++;
            }        
        }
        
        s = bitset<32>(count).to_string(); // count를 32bit의 2진수로 바꾸기 -> 문자열
        s = s.substr(s.find('1'));  
        
        answer[0]++; 
        answer[1] += n - count;
    }
    
    return answer;
}

File: Implementation/programmers/[Lv2] 이진 변환 반복하기/main.py
# Input
# s = 0과 1로 이루어진 문자열 
# Output
# 이진 변환 1. x의 모든 0을 제거 2. x의 길이 c일 때, x를 x를 2빈법으로 표현한 문자열
# s가 "1"이 될 때까지 이진 변환할 때 그 횟수와 제거된 0의 개수 
# Hint
# format(n, 'b') or f"{n:b}"
# 문자열의 서식 지정을 위해 사용함! 보통 숫자를 예쁘게 출력하기 위해 사용 

def solution(s):
    answer = [0, 0] 
    
    while s != "1":
        size = len(s)
        count = 0
        for z in s: # count = s.count('0')
            if z == "0": count += 1
            
        s = format(size-count, 'b')
        answer[0] += 1
        answer[1] += count
    
    return answer